draft-ietf-tls-esni.md

```diff
@@ -620,16 +623,27 @@ error code.
 
 ### HelloRetryRequest {#hrr}
 
+As required in {{send-ech}}, clients offering ECH MUST ensure that
+HelloRetryRequest-sensitive parameters applicable to TLS 1.3 or higher
+versions in ClientHelloInner match that in ClientHelloOuter. These parameters
+include:
```

Comment:
This connects parameters that might change in response to HRR to whether they appear inside or outside.  This assumes that they appear in both places (and that they are the same).

There's an implicit constraint here that probably needs to be explicit: HelloRetryRequest can only be sent to affect a change in a ClientHello that affects key negotiation.  This is recognizing a new constraint on the design that was essentially always true, but not mandatory.  In principle, prior to this, HRR could affect a new extension that didn't affect key negotiation.

Conveniently, we already have a proxy for that.  Extensions that are CH/SH (as opposed to CH/EE) are those that HRR can change.  In theory there could be some that appear in CH alone, but that is unlikely to be true for a new extension given our systemic rejection of trial decryption.

This should probably be reflected somewhere, probably by updating RFC 8446.

Reply:
I agree with the above, but it's not clear to me what specifically you're suggesting. The *intent* of this change is to encompass any CH/SH extension that might change due to HRR, and we refer to those as "HRR-sensitive" parameters. 

Can you please clarify what specifically what you're looking for, ideally with suggested text?

> This should probably be reflected somewhere, probably by updating RFC 8446.

I don't think this needs to update RFC8446. The HRR language in that document is somewhat unclear as-is, so it probably makes more sense to tackle this in rfc8446-bis.

Reply:
I think @martinthomson's point is that this PR is overly prescriptive about the semantics of HRR? If so, I disagree. It aims to address HRR-sensitive parameters that are known today, but there's line about how to address future changes to the HRR code path. Maybe we should be more upfront about this.